olive/engine/footprint.py

```diff
@@ -284,6 +284,35 @@ def plot_pareto_frontier(self, ranks=None, save_path=None, is_show=True, save_fo
         if is_show:
             fig.show()
 
+    def summarize_run_history(self, output_path):
+        """
+        Summarize the run history of a model with the columns of
+        model_id, parent_model_id, from_pass, duration, metrics
+        """
+        from tabulate import tabulate
+
+        headers = ["model_id", "parent_model_id", "from_pass", "duration_sec", "metrics"]
+        RunHistory = namedtuple("RunHistory", headers)
+        rls = []
+        for model_id, node in self.nodes.items():
+            # get the run duration between current model and its parent model
+            if node.parent_model_id is not None:
+                duration = max(node.end_time - node.start_time, 0)
+            else:
+                duration = None
+            run_history = RunHistory(
+                model_id=model_id,
+                parent_model_id=node.parent_model_id,
+                from_pass=node.from_pass,
+                duration_sec=duration,
+                metrics=str(node.metrics.value) if node.metrics else None,
+            )
+            rls.append(tuple(run_history))
+        tab_rls = tabulate(rls, headers=headers, tablefmt="grid")
+        logger.info(f"run history:\n{tab_rls}")
+        with open(output_path, "w") as f:
```

Comment:
could we return the result and let the caller decide whether to print or write to file?

Reply:
Updated.